WordPress/gutenberg
# Framework: Improve Travis build performance

This pull request seeks to explore a few ideas for improved Travis build performance:

- Remove `postinstall` script from `package.json`
   - Most of our Travis jobs ultimately call `npm run build` twice, wastefully, because of the combination of the `postinstall` triggered by an initial `npm install`, and a subsequent explicit build
   - `npm install` is run in many separate jobs, but `npm run check-licenses` only really needs to be run at most one time, not in each job
- Removes npm / JavaScript build from the "PHP Unit Tests" job
   - Avoids lengthy install / build process
   - As best I can tell, there should be no need to rely on JavaScript produced by `npm run build`. Previously it may have been needed to use the npm scripts for `npm run test-php` and `npm run test-unit-php-multisite`, but these resolve to `docker-compose` commands anyways, so it seems reasonable enough to call them directly
      - As of the current proposal of #13521, there is some PHP produced as the result of the Webpack build, though it is not subject to tests
- Runs "JS unit tests" job script tasks concurrently
   - With immediate bail-out if any of the tasks fail
   - None of these tasks has any dependency on any of the others
   - Arguably, this could make the build results even more difficult to interpret, since concurrent output will be mixed from each of the underlying tasks ([related Slack conversation](https://wordpress.slack.com/archives/C02QB2JS7/p1551274892295400)).

**Testing instructions:**

Verify a passing build.

In review, assure there is no lack of coverage for what had previously been tested.
